internal/activities/add_premis_agent.go

```diff
@@ -0,0 +1,51 @@
+package activities
+
+import (
+	"context"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/premis"
+)
+
+const AddPREMISAgentName = "add-premis-agent"
+
+type AddPREMISAgentParams struct {
+	PREMISFilePath string
+	Agent          premis.PREMISAgent
+}
+
+type AddPREMISAgentResult struct{}
+
+type AddPREMISAgentActivity struct{}
+
+func NewAddPREMISAgent() *AddPREMISAgentActivity {
+	return &AddPREMISAgentActivity{}
+}
+
+func (md *AddPREMISAgentActivity) Execute(
+	ctx context.Context,
+	params *AddPREMISAgentParams,
+) (*AddPREMISAgentResult, error) {
+	// Read existing XML, if any.
+	reader, err := premis.NewFileReaderIfFileExists(params.PREMISFilePath)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Always make sure to `defer reader.Close()` after opening a reader and checking the error.